front/pages/w/[wId]/assistant/new.tsx

```diff
@@ -170,287 +230,225 @@ export default function AssistantNew({
-          mentions: [{ configurationId: "helper" }],
-          contentFragments: [],
-        },
-        visibility: "test",
-      });
-      if (conversationRes.isErr()) {
-        if (conversationRes.error.type === "plan_limit_reached_error") {
-          setPlanLimitReached(true);
-        } else {
-          sendNotification({
-            title: conversationRes.error.title,
-            description: conversationRes.error.message,
-            type: "error",
-          });
-        }
-      } else {
-        setConversationHelperModal(conversationRes.value);
-      }
-    }
-  );
+  const {
+    metadata: quickGuideSeen,
+    isMetadataError: isQuickGuideSeenError,
+    isMetadataLoading: isQuickGuideSeenLoading,
+    mutateMetadata: mutateQuickGuideSeen,
+  } = useUserMetadata("quick_guide_seen");
 
+  const [showQuickGuide, setShowQuickGuide] = useState<boolean>(false);
   const [greeting, setGreeting] = useState<string>("");
-  const { animate, setAnimate, setSelectedAssistant } =
-    useContext(InputBarContext);
+
+  const { submit: handleCloseQuickGuide } = useSubmitFunction(async () => {
+    setUserMetadataFromClient({ key: "quick_guide_seen", value: "true" })
+      .then(() => {
+        return mutateQuickGuideSeen();
+      })
+      .catch(console.error);
+    setShowQuickGuide(false);
+  });
 
   useEffect(() => {
-    if (animate) {
-      setAnimate(false);
+    if (!quickGuideSeen && !isQuickGuideSeenError && !isQuickGuideSeenLoading) {
+      // Quick guide has never been shown, lets show it.
+      setShowQuickGuide(true);
     }
-  }, [animate, setAnimate]);
+  }, [isQuickGuideSeenError, isQuickGuideSeenLoading, quickGuideSeen]);
 
   useEffect(() => {
     setGreeting(getRandomGreetingForName(user.firstName));
   }, [user]);
 
-  const { submit: persistQuickGuideSeen } = useSubmitFunction(async () => {
-    setUserMetadataFromClient({ key: "quick_guide_seen", value: "true" })
-      .then(() => {
-        return mutateQuickGuideSeen();
-      })
-      .catch(console.error);
-    setShowQuickGuide(false);
+  const setInputbarMention = useCallback(
+    (agent: LightAgentConfigurationType) => {
+      setSelectedAssistant({
+        configurationId: agent.sId,
+      });
+      setAnimate(true);
+    },
+    [setSelectedAssistant, setAnimate]
+  );
+
+  const handleAssistantClick = useCallback(
+    // on click, scroll to the input bar and set the selected assistant
+    async (agent: LightAgentConfigurationType) => {
+      const scrollContainerElement = document.getElementById(
+        "assistant-input-header"
+      );
+
+      if (!scrollContainerElement) {
+        console.log("Unexpected: scrollContainerElement not found");
+        return;
+      }
+      const scrollDistance = scrollContainerElement.getBoundingClientRect().top;
+
+      // If the input bar is already in view (with a small margin of error), set the mention directly
+      if (-2 < scrollDistance) {
```

Comment:
Seen IRL, let's compare to 0, no need for an error margin here.

Reply:
Actually, it was needed :) Didn't manage to explain it properly IRL sorry about that

with the smooth scrolling you can arrive to a scrolltop of e.g. -0.34 . Then, if you click on the assistant, the condition does not fire, and there's no scrolling so no triggering of intersection observer either